src/Sylius/Bundle/AdminBundle/Resources/config/routing/ajax/product_variant.yml

```
@@ -37,6 +37,7 @@ sylius_admin_ajax_all_product_variants_by_phrase:
                 arguments:
                     phrase: $phrase
                     locale: expr:service('sylius.context.locale').getLocaleCode()
+                    limit: '!!int $limit'
```

It should be optional and have a default value, no need to require the user to pass it even if they're happy with the default value.
```suggestion
                    limit: '!!int $limit'
        limit: 25
```

---

Did you test it?
Inserting `limit: 25` below `defaults` denies possibility to pass `limit=123` in URL parameter (limit is always 25).

---

🤔 what...

---

Removing `, {limit: 25}` from `for_variants.html.twig` for this routing configuration:
```
defaults:
        _controller: sylius.controller.product_variant:indexAction
        _format: json
        _sylius:
            serialization_groups: [Autocomplete]
            permission: true
            repository:
                method: findByPhrase
                arguments:
                    phrase: $phrase
                    locale: expr:service('sylius.context.locale').getLocaleCode()
                    limit: '!!int $limit'
        limit: 2
```
and entering `/admin/ajax/product-variants/search-all?phrase=&limit=4` will return 2 variants in response.

---

You don't need to pass `limit` at all. That's why I added `0` check in repository method because empty limit becomes `0` instead of `null` which is imho not good but it's beyond this pr.

---

So, I had some time and I investigate the issue.
The problem is in the way resource bundle resolve those request parameters.
It uses [Request::get()](https://github.com/Sylius/SyliusResourceBundle/blob/8326f4c4b1daf6a430e242063d9f86988531520e/src/Bundle/Controller/ParametersParser.php#L64), which first looks in the request attributes bag.
Coincidentally, that's where the default route parameters values are stored.
This case proves that that logic gives a bad DX and should be changed, just like Symfony recommends it.
So, here's my solution around it (that I also tested):
```
                    limit: "expr:service('request_stack').getCurrentRequest().query.getInt('limit') > 0 ? service('request_stack').getCurrentRequest().query.getInt('limit') : null"
        limit: 25
```
This will also solve the issue with null being converted to 0 and you will have either a null or integer sent to the repository method, so you can probably remove the condition there.

/cc @loic425 since we talked about that recently

---

What is that second limit key for?

---

In the expression or the default `limit: 25`?

---

Hopefully we can come up with a better solution in the resource bundle, so things like that are easier to do.